core/src/main/java/tc/oc/pgm/spawner/SpawnerModule.java

```diff
@@ -49,58 +54,86 @@ public SpawnerModule parse(MapFactory factory, Logger logger, Document doc)
       RegionParser regionParser = factory.getRegions();
       KitParser kitParser = factory.getKits();
       FilterParser filterParser = factory.getFilters();
+      AtomicInteger numericId = new AtomicInteger(0);
 
-      int numericID = 0;
-      for (Element element :
+      for (Element spawnerEl :
           XMLUtils.flattenElements(doc.getRootElement(), "spawners", "spawner")) {
-        Region spawnRegion = regionParser.parseRequiredRegionProperty(element, "spawn-region");
-        Region playerRegion = regionParser.parseRequiredRegionProperty(element, "player-region");
-        Attribute delayAttr = element.getAttribute("delay");
-        Attribute minDelayAttr = element.getAttribute("min-delay");
-        Attribute maxDelayAttr = element.getAttribute("max-delay");
+        String id = spawnerEl.getAttributeValue("id");
+        Region spawnRegion = regionParser.parseRequiredRegionProperty(spawnerEl, "spawn-region");
+        Region playerRegion = regionParser.parseRequiredRegionProperty(spawnerEl, "player-region");
+        Attribute delayAttr = spawnerEl.getAttribute("delay");
+        Attribute minDelayAttr = spawnerEl.getAttribute("min-delay");
+        Attribute maxDelayAttr = spawnerEl.getAttribute("max-delay");
+
+        if (spawnerEl.getAttributeValue("id") == null) {
+          id = "spawner-" + numericId.getAndIncrement();
```

Comment:
move the generation of this id to SpawnerDefinition itself, check PostDefinition for an example
Also use pgm's default fallback generation, iirc it goes a bit like `--(type)-(name)` in this case type would be spawner and name just the id (like posts) so: `--spawner-1` and such
It generally will generate less collisions on accident